### ISSUE TITLE
Fastly serving cached pages in maintenance mode - whitelisted IP

### ISSUE THREAD

**Issue by molsm** [fastly/fastly-magento2#183]
Hello Fastly team,

## Summary of issue

We have acquired strange behaviour. Fastly services was able to cache pages during maintenance mode who have list of whitelisted IP ( var/.maintenance.ip ) .

Once whitelisted user (admin) hit certain pages, Fastly cached them and after that was able serve actual page to user who has been entered from non-whitelisted IP (out of network).

## Steps to replicate

1. Setup Fastly (latest version) on Magento 2.2.3 
2. Make sure to add your IP in var/.maintenance.ip
3. Put page in maintenance mode
4. Check homepage, if it is able to serve from outer network
5. Hit homepage from whitelisted IP
6. Check homepage from outer network and receive cached page

## Expected outcome

User from outer network should receive maintenance page instead of cached page (by whitelisted user)

## Questions

1. This is bug can be reproduced only on our side?
2. Can Fastly module implemented correct behaviour in this case? We already working on workaround but it is unhandy.

Best regards,

**Comment by vvuksan** [comment]
Unfortunately this is expected behavior at this time as the module has no notion of var/maintenance.ip. It would need to create an ACL. This has been added as a feature request.

**Comment by molsm** [comment]
Hello @vvuksan ,

We have implemented solution. Are you interested to get more details or contribution PR?

Best regards,

**Comment by vvuksan** [comment]
Yes please submit a PR.

Thank you

**Comment by vvuksan** [comment]
@molsm are you able to submit a PR for this ?

**Comment by molsm** [comment]
Hello @vvuksan ,

I apologize, will not able to make PR now, because I am tight on time. 

But I can describe what we have done on our project.

Implementation was pretty straightforward:

1. Create Edge ACL with whitelisted IP's - maint_allowlist
2. Create Edge Dictionary "configtable" with key-value pair - allow_super_users_during_maint -> 1/0
3. Uploaded and activated custom snippet maintenance.json

```
"content": "if ( table.lookup(configtable, \"allow_super_users_during_maint\", \"0\") == \"1\" && !req.http.Fastly-Client-IP ~ maint_allowlist ) { set req.http.ResponseObject = \"970\"; }"
```

4. Implemented PHP API class that can update dictionary `allow_super_users_during_maint` item with 0 or 1
5. Operate with PHP Api class through CLI commands `php bin/magento fastly:maintenance:enable`

Best regards,

**Comment by vvuksan** [comment]
Fixed in #256
